libraries/integration/Microsoft.Bot.Builder.ApplicationInsights.Core/TelemetryBotIdInitializer.cs

```diff
@@ -16,44 +15,51 @@ namespace Microsoft.Bot.Builder.ApplicationInsights.Core
     /// </summary>
     public class TelemetryBotIdInitializer : ITelemetryInitializer
     {
-        private readonly IMemoryCache _memoryCache;
+        public static readonly string BotActivityKey = "BotActivity";
         private readonly IHttpContextAccessor _httpContextAccessor;
 
-        public TelemetryBotIdInitializer(IHttpContextAccessor httpContextAccessor, IMemoryCache memoryCache)
+        public TelemetryBotIdInitializer(IHttpContextAccessor httpContextAccessor)
         {
-            _memoryCache = memoryCache ?? throw new ArgumentNullException(nameof(memoryCache));
             _httpContextAccessor = httpContextAccessor ?? throw new ArgumentNullException(nameof(httpContextAccessor));
         }
 
         public void Initialize(ITelemetry telemetry)
         {
-            var context = _httpContextAccessor.HttpContext;
-            if (context != null && (telemetry is RequestTelemetry || telemetry is EventTelemetry))
+            if (telemetry == null)
```

Comment:
I guess if you want to be super safe you can check, but I don't think this is possible by definition of the contract of the interface that you would ever get a `null` value here.